pfsimulator/parflow_lib/solver_richards.c

```diff
@@ -2885,6 +2888,69 @@ AdvanceRichards(PFModule * this_module, double start_time,      /* Starting time
       }
     }
 
+
+   /***************************************************************
+    *          modify land surface pressures                      *
+    ***************************************************************/
+   
+    if (public_xtra->reset_surface_pressure == 1)
+    {
+      GrGeomSolid *gr_domain = ProblemDataGrDomain(problem_data);
+
+      int i, j, k, r, is;
+      int ix, iy, iz;
+      int nx, ny, nz;
+      int ip;
+      // JLW add declarations for use without CLM
+      Subvector *p_sub_sp;
+      double *pp_sp;
+
+      Subgrid *subgrid;
+      Grid *grid = VectorGrid(evap_trans_sum);
+
+      ForSubgridI(is, GridSubgrids(grid))
+      {
+        subgrid = GridSubgrid(grid, is);
+        p_sub_sp = VectorSubvector(instance_xtra->pressure, is);
+
+        r = SubgridRX(subgrid);
+
+        ix = SubgridIX(subgrid);
+        iy = SubgridIY(subgrid);
+        iz = SubgridIZ(subgrid);
+
+        nx = SubgridNX(subgrid);
+        ny = SubgridNY(subgrid);
+        nz = SubgridNZ(subgrid);
+
+        pp_sp = SubvectorData(p_sub_sp);
+
+        GrGeomInLoop(i, j, k, gr_domain, r, ix, iy, iz, nx, ny, nz,
+        {
+          ip = SubvectorEltIndex(p_sub_sp, i, j, k);
+          // printf(" %d %d %d %d  \n",i,j,k,ip);
+          // printf(" pp[ip] %10.3f \n",pp[ip]);
+          // printf(" NZ: %d \n",nz);
+          if (k == (nz - 1))
+          {
+            //   printf(" %d %d %d %d  \n",i,j,k,ip);
+            //   printf(" pp[ip] %10.3f \n",pp[ip]);
+
+            if (pp_sp[ip] > public_xtra->threshold_pressure)
+            {
+              printf(" pressure reset: %d %d %d %10.3f \n", i, j, k,
```

Comment:
One of the recommend practices is to not have output to stdout.   Use amps_Printf instead of printf so we can control output destination.

Is the current timestep useful in this diagnostic message?

Reply:
Thanks @smithsg84 that is a very good catch, please do make this change.  And yes, the current timestep info would be really helpful here.